crates/hir_ty/src/diagnostics.rs

```diff
@@ -318,6 +318,31 @@ impl Diagnostic for MismatchedArgCount {
     }
 }
 
+#[derive(Debug)]
+pub struct AddReferenceToArg {
+    pub file: HirFileId,
+    pub arg_expr: AstPtr<ast::Expr>,
+    pub mutability: Mutability,
+}
```

Comment:
@jonas-schievink do we need to do anything special to make this diagnostic as experimental?

Reply:
(yeah, added the necessary code as a suggestion)